vpplink/interfaces.go

```diff
@@ -95,19 +94,7 @@ func (v *VppLink) SetInterfaceMacAddress(swIfIndex uint32, mac *net.HardwareAddr
 	return nil
 }
 
-func (v *VppLink) SetInterfaceVRF(swIfIndex, vrfIndex uint32) error {
-	err := v.SetInterfaceVRFAf(swIfIndex, vrfIndex, false)
-	if err != nil {
-		return err
-	}
-	err = v.SetInterfaceVRFAf(swIfIndex, vrfIndex, true)
-	if err != nil {
-		return err
-	}
-	return nil
-}
-
-func (v *VppLink) SetInterfaceVRFAf(swIfIndex, vrfIndex uint32, isIP6 bool) error {
+func (v *VppLink) SetInterfaceVRF(swIfIndex, vrfIndex uint32, isIP6 bool) error {
```

Comment:
Why remove the helper that does 2 AFs at once?

Reply:
I tried with the helpers, doing two AFs at once.
But the issue (generally speaking) comes with error handling.
In the caller you don't know which call generated the error, and if you need to cleanup that gets quickly ugly.

The solution I ended up with is something like :
````go
for _, ipFamily := range vpplink.IpFamilies {
	err = s.vpp.AddDefaultRouteViaTable(..., ipFamily.IsIp6)
	if err != nil {
		return errors.Wrapf(err, "error for %s", ipFamily.Str)
	} else {
		stack.Push(s.vpp.DelDefaultRouteViaTable, ..., ipFamily.IsIp6)
	}
}
````

Reply:
Makes sense :)